src/GuiBase/Viewer/Gizmo/RotateGizmo.cpp

```diff
@@ -60,14 +53,14 @@ RotateGizmo::RotateGizmo( Engine::Component* c,
         Engine::RenderObject* arrowDrawable =
             new Engine::RenderObject( "Gizmo Torus", m_comp, Engine::RenderObjectType::UI );
 
-        std::shared_ptr<Engine::RenderTechnique> rt( new Engine::RenderTechnique );
-        rt->setConfiguration( Ra::Engine::ShaderConfigurationFactory::getConfiguration( "Plain" ) );
-        rt->resetMaterial( new Ra::Engine::BlinnPhongMaterial( "Default material" ) );
+        std::shared_ptr<Engine::RenderTechnique> rt(
+            makeRenderTechnique( "Rotate Gizmo material",
+                                 mesh->getCoreGeometry().hasAttrib( Engine::Mesh::getAttribName(
+                                     Engine::Mesh::VERTEX_COLOR ) ) ) );
```

Comment:
As mentioned above, update to the new PlainMaterial design and use perVertexColor only where needed.
It would mean having 4 different Materials, one per axis ( red, green blue) + one for selection (yellow), and changing the material on the fly rather than changing all the vertices colors.
Maybe the Materials could be static members of the `Gizmo` class.
Also, the `RenderTechnique` effectively depends on the mesh, but they only differ through the vertex colors and this oesn't affect the `RenderTechnique`.

Reply:
Not sure how to do it, but you still have access to ShaderParameterProvider, then cast to plainmaterial and change pervertexcolor

Reply:
I would rather change the material from R/G/B to Y, then back from Y to R/G/B/